msbuild/Test.README.md

```diff
@@ -0,0 +1,172 @@
+
+:white_check_mark: A build engine is needed, a class which implements [IBuildEngine](https://docs.microsoft.com/dotnet/api/microsoft.build.framework.ibuildengine?view=msbuild-17-netcore). In our example we created a mock using [Moq](https://github.com/Moq/moq4/wiki/Quickstart), but you can use other mock tools. I was interesting in collecting the errors, but you can collect other information and then assert it.
+The Engine Mock is needed on all the tests, so it was included as _TestInitialize_ (it is going to be executed before each test, and each test will have its own build engine). [Complete example](./custom-task-code-generation/AppSettingStronglyTyped/AppSettingStronglyTyped.Test/AppSettingStronglyTypedTest.cs)
+
+```c#
+       private Mock<IBuildEngine> buildEngine;
+       private List<BuildErrorEventArgs> errors;
+
+        [TestInitialize()]
+        public void Startup()
+        {
+            buildEngine = new Mock<IBuildEngine>();
+            errors = new List<BuildErrorEventArgs>();
+            buildEngine.Setup(x => x.LogErrorEvent(It.IsAny<BuildErrorEventArgs>())).Callback<BuildErrorEventArgs>(e => errors.Add(e));
+        }
+
+```
+
+Now we need to create our Task and set the parameters as part of the test arrangement.
+
+```csharp
+   //Arrange
+   var item = new Mock<ITaskItem>();
+   item.Setup(x => x.GetMetadata("Identity")).Returns($".\\Resources\\complete-prop.setting");
+
+   var appSettingStronglyTyped = new AppSettingStronglyTyped { SettingClassName = "MyCompletePropSetting", SettingNamespaceName = "MyNamespace", SettingFiles = new[] { item.Object } };
+
+   appSettingStronglyTyped.BuildEngine = buildEngine.Object;
+```
+
+First, we create the ITaskItem parameter mock (using [Moq](https://github.com/Moq/moq4/wiki/Quickstart)), and point to the file to be parsed. Then, we create our _AppSettingStronglyTyped_ Custom Task with its parameters. Finally, we set the build engine to our MSBuild Custom Task.
+
+At this point we need to do the action
+
+```csharp
+    //Act
+    var success = appSettingStronglyTyped.Execute();
+```
+
+Last but not least, we need to assert the expected outcome from our test
+
+```csharp
+   //Assert
+   Assert.IsTrue(success); // The execution was success
+   Assert.AreEqual(errors.Count, 0); //Not error were found
+   Assert.AreEqual($"MyCompletePropSetting.generated.cs", appSettingStronglyTyped.ClassNameFile); // The Task expected output
+   Assert.AreEqual(true, File.Exists(appSettingStronglyTyped.ClassNameFile)); // The file was generated
+   Assert.IsTrue(File.ReadLines(appSettingStronglyTyped.ClassNameFile).SequenceEqual(File.ReadLines(".\\Resources\\complete-prop-class.txt"))); // Assenting the file content
+```
+
+Following this pattern you should expand all the possibilities.
+:warning: When there are files generation we need to use different file name for each test to avoid collision. Remember to delete the generated files as test cleanup.
```

Comment:
```suggestion
:warning: When there are files generated, we need to use different file name for each test to avoid collision. Remember to delete the generated files as test cleanup.
```